pybamm/solvers/c_solvers/idaklu/IDAKLUSolverOpenMP.inl

```diff
@@ -375,63 +463,62 @@ Solution IDAKLUSolverOpenMP<ExprSet>::solve(
       const int base_index = j * number_of_timesteps * number_of_states;
-      for (int k = 0; k < number_of_states; k++)
+      for (int k = 0; k < number_of_states; k++) {
         yS_return[base_index + k] = ySval[j][k];
+      }
     }
   }
 
   // Subsequent states (t_i>0)
   int retval;
   t_i = 1;
-  while (true)
-  {
+  while (true) {
     realtype t_next = t(t_i);
     IDASetStopTime(ida_mem, t_next);
     DEBUG("IDASolve");
     retval = IDASolve(ida_mem, t_final, &tret, yy, yp, IDA_NORMAL);
 
-    if (retval == IDA_TSTOP_RETURN ||
+    if (!(retval == IDA_TSTOP_RETURN ||
         retval == IDA_SUCCESS ||
-        retval == IDA_ROOT_RETURN)
-    {
-      if (number_of_parameters > 0)
-        IDAGetSens(ida_mem, &tret, yyS);
-
-      // Evaluate and store results for the time step
-      t_return[t_i] = tret;
-      if (functions->var_fcns.size() > 0) {
-        // Evaluate functions for each requested variable and store
-        // NOTE: Indexing of yS_return is (time:var:param)
-        CalcVars(y_return, length_of_return_vector, t_i,
-                 &tret, yval, ySval, yS_return, &ySk);
-      } else {
-        // Retain complete copy of the state vector
-        for (int j = 0; j < number_of_states; j++)
-          y_return[t_i * number_of_states + j] = yval[j];
-        for (int j = 0; j < number_of_parameters; j++)
-        {
-          const int base_index =
-            j * number_of_timesteps * number_of_states +
-            t_i * number_of_states;
-          for (int k = 0; k < number_of_states; k++)
-            // NOTE: Indexing of yS_return is (time:param:yvec)
-            yS_return[base_index + k] = ySval[j][k];
+        retval == IDA_ROOT_RETURN)) {
+      // failed
+      break;
+    }
+
+    if (sensitivity) {
+      CheckErrors(IDAGetSens(ida_mem, &tret, yyS));
+    }
+
+    // Evaluate and store results for the time step
+    t_return[t_i] = tret;
+    if (functions->var_fcns.size() > 0) {
+      // Evaluate functions for each requested variable and store
+      // NOTE: Indexing of yS_return is (time:var:param)
+      CalcVars(y_return, length_of_return_vector, t_i,
+                &tret, yval, ySval, yS_return, &ySk);
+    } else {
+      // Retain complete copy of the state vector
+      for (int j = 0; j < number_of_states; j++) {
+        y_return[t_i * number_of_states + j] = yval[j];
+      }
+      for (int j = 0; j < number_of_parameters; j++) {
+        const int base_index =
+          j * number_of_timesteps * number_of_states +
+          t_i * number_of_states;
+        for (int k = 0; k < number_of_states; k++) {
+          // NOTE: Indexing of yS_return is (time:param:yvec)
+          yS_return[base_index + k] = ySval[j][k];
         }
       }
-      t_i += 1;
-
-      if (retval == IDA_SUCCESS ||
-          retval == IDA_ROOT_RETURN)
-        break;
     }
-    else
-    {
-      // failed
+    t_i += 1;
+
+    if (retval == IDA_SUCCESS || retval == IDA_ROOT_RETURN) {
```

Comment:
should IDA_TSTOP_RETURN be in here as well? (I might be wrong, getting confused trying to follow control flow via the diff...)

Reply:
The logic inside the while loop is identical to how it was before. My only update was rearranging the if statement from
```cpp
if (success) {
   ...
} else {
   break
}
```
to
```cpp
if (!success) {
   break
}
...
```
The large diff is because of indentation.